include/fmt/printf.h

```diff
@@ -517,10 +519,11 @@ OutputIt basic_printf_context<OutputIt, Char>::format() {
     if (specs.alt && visit_format_arg(detail::is_zero_int(), arg))
       specs.alt = false;
     if (specs.fill[0] == '0') {
-      if (arg.is_arithmetic())
+      if (arg.is_arithmetic() && specs.align != align::left)
         specs.align = align::numeric;
       else
-        specs.fill[0] = ' ';  // Ignore '0' flag for non-numeric types.
+        specs.fill[0] = ' ';  // Ignore '0' flag for non-numeric types or if '-'
+                              // flag is also present.
```

Comment:
opengroup description: "If the `0` and `-` flags both appear, the `0` flag is ignored."